core_classes.py

```diff
@@ -0,0 +1,343 @@
+import pandas as pd 
+import numpy as np 
+
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+'''
+class class_name(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        
+        # add the necessary attributes 
+        
+    
+    def fit(self,x,y=None):
+        
+        # learn from the data 
+        # populate attributes 
+        # learn featurenames 
+
+        return self 
+    
+    def transform(self,x) :
+        
+        # make use of what you learnt in fit 
+        # or othwerwise transform/modify dat
+
+        return modified_data
+
+
+        
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+'''
+
+class mytfidf(BaseEstimator, TransformerMixin):
+
```

Comment:
this is not written to handle multiple text columns and we have to assume that it gets only a single text column from the upstream processes in the pipeline .